astropy/table/table.py

```diff
@@ -675,7 +675,7 @@ def _convert_col_for_table(self, col):
         MaskedColumn if the table is masked.  Table subclasses like QTable
         override this method.
         """
-        if isinstance(col, Column) and not col.__class__ is self.ColumnClass:
```

Comment:
Huh. Is the second comparison not equivalent to `not isinstance(col, self.ColumnClass)`?